lib/liquid/context.rb

```diff
@@ -176,7 +180,7 @@ def []=(key, value)
     # Example:
     #   products == empty #=> products.empty?
     def [](expression)
-      evaluate(Expression.parse(expression))
+      evaluate(Expression.parse(expression, @string_scanner))
```

Comment:
Side note: I think `Context#[]` should just be deprecated honestly, it's a very deceptive performance footgun.

Instead, we can add a new method: `Context#parse_and_evaluate`